multiapps-mta/src/test/java/org/cloudfoundry/multiapps/mta/model/VersionTest.java

```diff
@@ -0,0 +1,145 @@
+package org.cloudfoundry.multiapps.mta.model;
+
+import java.text.MessageFormat;
+
+import org.cloudfoundry.multiapps.common.ParsingException;
+import org.cloudfoundry.multiapps.mta.Messages;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class VersionTest {
+
+    @Nested
+    @DisplayName("parseVersion() tests")
+    class ParseVersionTests {
+
+        @Test
+        @DisplayName("should parse a standard semantic version successfully")
+        void testParseStandardVersion() {
+            Version version = Version.parseVersion("1.2.3");
+
+            assertEquals(1, version.getMajor());
+            assertEquals(2, version.getMinor());
+            assertEquals(3, version.getPatch());
+            assertEquals("1.2.3", version.toString());
+        }
+
+        @Test
+        @DisplayName("should parse versions with build metadata and timestamps")
+        void testParseComplexVersions() {
+            Version v1 = Version.parseVersion("0.0.1-20251105063220+e303076b5d51da7f0f3a10cd69de018ac0a3853d");
+            Version v2 = Version.parseVersion("1.2.0-20251105043948+cb754700fb069b3367c869938ac6beb396c800e4");
+            Version v3 = Version.parseVersion("1.0.0-SNAPSHOT-20240116224612+200df4832787863cc2f94d998c6cbcd711518933");
+
+            assertNotNull(v1);
+            assertNotNull(v2);
+            assertNotNull(v3);
+
+            // Check major/minor/patch extracted properly
+            assertEquals(0, v1.getMajor());
```

Comment:
Add assertions for all variables and for three methods - getMajor, getMinor and getPatch. You can create such method and reuse it for all Version object asserts.